client/repoutils.c

```diff
@@ -553,18 +555,23 @@ TDNFFindRepoById(
         dwError = ERROR_TDNF_NO_REPOS;
         BAIL_ON_TDNF_ERROR(dwError);
     }
-    pRepos = pTdnf->pRepos;
+    parse_pRepos = pTdnf->pRepos;
 
-    while(pRepos)
+    while(parse_pRepos)
     {
-        if(!strcmp(pszRepo, pRepos->pszId))
+        if(!strcmp(pszRepo, parse_pRepos->pszId) && nFound == 0)
         {
-            break;
+            nFound = 1;
+            pRepos = parse_pRepos;
+        } else if(!strcmp(pszRepo, parse_pRepos->pszId) && nFound == 1)
+        {
+            dwError = ERROR_TDNF_DUPLICATE_REPO_ID;
+            BAIL_ON_TDNF_ERROR(dwError);
         }
-        pRepos = pRepos->pNext;
+        parse_pRepos = parse_pRepos->pNext;
     }
```

Comment:
This is the wrong place for this check, for two reasons: this function will be called too late, or may actually not be called at all. It's utility function to find a repo, it's not intended to check the configuration. So it may also be called multiple times, and since this adds extra CPU cycles, makes it inefficient.

A better place for the check would be in `TDNFLoadRepoData()` in `client/repolist.c` at the end. You can loop through all repos there and check for dups.

Besides all that, this logic also has a flaw - we'd only detect duplicates for repositories we are looking for. If for example we are looking for `photon-bar`, and `photon-foo` is a dup, we wouldn't see that.